hatrac/model/storage/amazons3.py

```diff
@@ -228,3 +232,49 @@ def delete_namespace(self, name):
         """Tidy up after an empty namespace that has been deleted."""
         # nothing to do for S3 since namespaces are not explicit resources in bucket
         pass
+
+class InputWrapper():
+    """Input stream file-like wrapper for uploading data to S3. 
+
+    This module wraps mod_wsgi_input providing implementations of
+    seek and tell that are used by boto (but not relied upon)
+
+    """
+
+    def __init__(self, ip, nbytes):
+        self._mod_wsgi_input = ip
+        self.nbytes = nbytes
+        self.reading_started = False
+        self.current_position = 0
+
+    def read(self, size=None):
+        if self.current_position != 0:
+            raise Exception("Stream seek position not at 0")
+        self.reading_started = True
+        return self._mod_wsgi_input.read(size)
+
+    def tell(self):
+        if self.reading_started: 
+            raise Exception("Stream reading started")
+        
+        return self.current_position
+
+    def seek(self, offset, whence=0):
+        if self.reading_started: 
+            raise Exception("Stream reading started")
+        
+        if whence == 0:
+            if offset > self.nbytes or offset < 0:
+                raise IOException("Can't seek beyond stream lenght")
+            self.current_position = offset
+        elif whence == 1:
+            if offset + self.current_position > self.nbytes or offset + self.current_position < 0:
+                raise IOException("Can't seek beyond stream length") 
+            self.current_position = self.current_position + offset
+        else:
+            if offset > 0 or self.nbytes + offset < 0:
+                raise IOException("Can't seek beyond stream length")
+            self.current_position = self.nbytes + offset
+             
+    def name(self):
+        return self._mod.wsgi_input.name()
```

Comment:
This should be `self._mod_wsgi_input.name()` not `self._mod.wsgi_input.name()`, right?